File: src/main/java/com/paragon/client/systems/module/impl/misc/XCarry.java
package com.paragon.client.systems.module.impl.misc;

import com.paragon.api.event.network.PacketEvent;
import com.paragon.asm.mixins.accessor.ICPacketCloseWindow;
import com.paragon.client.systems.module.Module;
import com.paragon.client.systems.module.Category;
import me.wolfsurge.cerauno.listener.Listener;
import net.minecraft.network.play.client.CPacketCloseWindow;

/**
 * @author Wolfsurge
 */
public class XCarry extends Module {

    public XCarry() {
        super("XCarry", Category.MISC, "Lets you carry items in your crafting grid");
    }

    @Listener
    public void onPacketSent(PacketEvent.PreSend event) {
        if (event.getPacket() instanceof CPacketCloseWindow) {
            if (((ICPacketCloseWindow) event.getPacket()).getID() == mc.player.inventoryContainer.windowId) {
                event.cancel();
            }
        }
    }

}

File: src/main/kotlin/com/paragon/client/systems/module/impl/misc/XCarry.kt
package com.paragon.client.systems.module.impl.misc

import com.paragon.api.event.network.PacketEvent.PreSend
import com.paragon.api.module.Category
import com.paragon.api.module.Module
import com.paragon.mixins.accessor.ICPacketCloseWindow
import com.paragon.bus.listener.Listener
import net.minecraft.network.play.client.CPacketCloseWindow

/**
 * @author Surge
 */
object XCarry : Module("XCarry", Category.MISC, "Lets you carry items in your crafting grid") {

    @Listener
    fun onPacketSent(event: PreSend) {
        if (event.packet is CPacketCloseWindow && (event.packet as ICPacketCloseWindow).id == minecraft.player.inventoryContainer.windowId) {
            event.cancel()
        }
    }

}